### PR TITLE
idle and attack sfx

### PR DESCRIPTION
**Pull request by AlecatEnjoyer**
This branch include the idle sfx and attack sfx for the limbs of the monster. They are also called attack specific moments of spawning or attacking.

### CODE REVIEW

File: Assets/GameAssets/Boss/BossScripts/Attacks/Vine/VineAttack.cs [hunk 1]
```diff
@@ -59,6 +59,7 @@ private void SetStartingValues(VineAttackController vineAttackController)
     public void SpawnVine()
     {
         _spawnedVine = Instantiate(_vineAttackPrefab, _vineAttackSpawnLocation);
+
```
Comment (astapay) [inline Assets/GameAssets/Boss/BossScripts/Attacks/Vine/VineAttack.cs:62]: Empty line that may need to be removed.

File: Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs [hunk 2]
```diff
@@ -48,6 +53,40 @@ protected virtual void BeginAttack()
         _isAttackActive = true;
     }
 
+    protected virtual void PlayIdleLoop(float second)
```
Comment (RyanASwanson) [inline Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs:56]: XML comment

File: Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs [hunk 3]
```diff
@@ -48,6 +53,40 @@ protected virtual void BeginAttack()
...
+    {
+        if (_idleSFXCoroutine == null)
```
Comment (RyanASwanson) [inline Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs:58]: Is this supposed to return if null? There is no return

File: Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs [hunk 4]
```diff
@@ -48,6 +53,40 @@ protected virtual void BeginAttack()
...
+        _idleSFXCoroutine = StartCoroutine(LoopIdleCorrectfloat(second));
+    }
+    protected virtual void DestroyIdleLoop()
```
Comment (RyanASwanson) [inline Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs:61]: XML comment and add a space between the 2 functions

**Review by astapay — APPROVED** [review]
Found an empty line in a method. approved

File: Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs [hunk 5]
```diff
@@ -48,6 +53,40 @@ protected virtual void BeginAttack()
...
+    {
+        if (_idleSFXCoroutine != null)
+        {
+            StopCoroutine(_idleSFXCoroutine);
+        }
+    }
```
Comment (RyanASwanson) [inline Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs:67]: Add space after this function

File: Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs [hunk 6]
```diff
@@ -48,6 +53,40 @@ protected virtual void BeginAttack()
...
+    /// <summary>
+    /// Play heartbeat at controlled rate 
+    /// </summary>
+    /// <returns></returns>
+    private IEnumerator LoopIdleCorrectfloat (float second)
```
Comment (RyanASwanson) [inline Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs:72]: This name is confusing to me. Also the F in float should be uppercase, but if you rename it then nvm. Perhaps we could get a more descriptive name than seconds. Maybe loopTime?

File: Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs [hunk 7]
```diff
@@ -48,6 +53,40 @@ protected virtual void BeginAttack()
...
+    protected virtual void DestroyIdleLoop()
+    {
+        if (_idleSFXCoroutine != null)
+        {
+            StopCoroutine(_idleSFXCoroutine);
+        }
+    }
+    /// <summary>
+    /// Play heartbeat at controlled rate 
+    /// </summary>
```
Comment (RyanASwanson) [inline Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs:70]: XML comment is missing the float variable

File: Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs [hunk 8]
```diff
@@ -48,6 +53,40 @@ protected virtual void BeginAttack()
...
+        float timer = 0.0f;
+
+        while (true)
+        {
+            if (timer > second)
+            {
+                RuntimeSfxManager.APlayOneShotSfx?.Invoke(FmodSfxEvents.Instance.LimbIdle, transform.position);
+                timer = 0.0f;
+            }
+            timer += Time.deltaTime;
```
Comment (RyanASwanson) [inline Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs:83]: Could we do just do a WaitForSeconds instead?

File: Assets/GameAssets/Boss/BossScripts/Attacks/Lockdown/LockdownAttackController.cs [hunk 9]
```diff
@@ -24,6 +24,10 @@ public class LockdownAttackController : BaseBossAttack
     [Tooltip("The core of the patrol enemy. The part that receives damage")]
     [SerializeField] private GameObject _coreEnemyPrefab;
 
+    [Space]
```
Comment (RyanASwanson) [inline Assets/GameAssets/Boss/BossScripts/Attacks/Lockdown/LockdownAttackController.cs:27]: Add self as contributor to this script

File: Assets/GameAssets/Boss/BossScripts/Attacks/Vine/VineAttack.cs [hunk 10]
```diff
@@ -59,6 +59,7 @@ private void SetStartingValues(VineAttackController vineAttackController)
     public void SpawnVine()
     {
         _spawnedVine = Instantiate(_vineAttackPrefab, _vineAttackSpawnLocation);
+        
     }
```
Comment (RyanASwanson) [inline Assets/GameAssets/Boss/BossScripts/Attacks/Vine/VineAttack.cs:64]: Add self as contributor to this script

File: Assets/GameAssets/Boss/BossScripts/Attacks/Vine/VineAttackController.cs [hunk 11]
```diff
@@ -23,7 +23,8 @@ public class VineAttackController : BaseBossAttack
     /// </summary>
```
Comment (RyanASwanson) [inline Assets/GameAssets/Boss/BossScripts/Attacks/Vine/VineAttackController.cs:23]: Add self as contributor to this script

File: Assets/General/Audio/FMODScripts/FmodSfxEvents.cs [hunk 12]
```diff
@@ -35,7 +36,7 @@ public class FmodSfxEvents : MonoBehaviour
```
Comment (RyanASwanson) [inline Assets/General/Audio/FMODScripts/FmodSfxEvents.cs:36]: Add self as contributor to this script

**Review by RyanASwanson — CHANGES_REQUESTED** [review]
Could use some adjustments to better follow code standards

File: Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs (hunk 3 above)
Comment (AlecatEnjoyer) [reply Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs:58]: not really just for starting the coroutine but I can add that

File: Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs (hunk 8 above)
Comment (AlecatEnjoyer) [reply Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs:83]: yes :)

File: Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs (hunk 7 above)
Comment (AlecatEnjoyer) [reply Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs:81]: got it now

File: Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs [hunk 13]
```diff
@@ -52,41 +52,44 @@ protected virtual void BeginAttack()
...
         {
-            if (timer > second)
-            {
-                RuntimeSfxManager.APlayOneShotSfx?.Invoke(FmodSfxEvents.Instance.LimbIdle, transform.position);
-                timer = 0.0f;
-            }
-            timer += Time.deltaTime;
-
-            yield return null;
+            yield return new WaitForSeconds(second);
```
[the remark this answers is not in the document]
Comment (AlecatEnjoyer) [reply Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs:87]: optimized

**Pull request merged**